components/raftstore/src/store/peer.rs

```diff
@@ -1378,17 +1387,27 @@ where
     fn on_leader_changed<T>(
         &mut self,
         ctx: &mut PollContext<EK, ER, T>,
-        leader_id: u64,
-        term: u64,
+        leader_id: Option<u64>,
+        term: Option<u64>,
     ) {
+        let mut meta = ctx.store_meta.lock().unwrap();
+        match meta.leaders.entry(self.region_id) {
+            HashEntry::Occupied(mut e) => {
+                let term = term.unwrap_or_else(|| e.get().0);
+                let leader_id = leader_id.unwrap_or_else(|| e.get().1);
```

Comment:
I don't think it's correct. term and leader id can't be mixed arbitrary.

Reply:
It just means (term, leader id) could be updated partly.

Reply:
That's exactly what I mean. For example, (4, 5) can't be updated to (5, 5). Because 5 is leader at term 4 but may not be leader at term 5.

Reply:
Ideally, leader id and term should be updated together with a yielded `SoftState`, but sometimes the leader was transferred to another store and then changed back, a follower would not get a new `SoftState`. In this situation, we can only update the term.

Reply:
Then follower will also record the leader id in soft state, you can query it by `self.leader_id()`.